src/content/docs/en/guides/upgrade-to/v3.mdx

```diff
@@ -0,0 +1,529 @@
+
+```js title="astro.config.mjs" del={4} ins={5}
+import { defineConfig } from "astro/config";
+
+export default defineConfig({
+  base: 'my-base',
+  base: 'my-base/',
+});
+```
+
+### Changed default: `compressHTML`
+
+In Astro v2.x, Astro only compressed your emitted HTML when [`compressHTML`](/en/reference/configuration-reference/#compresshtml) was explicitly set to `true`. The default value was `false`.
+
+Astro v3.0 now compresses emitted HTML by default.
+
+#### What should I do?
+
+You can now remove `compressHTML: true` from your configuration as this is the new default behavior.
+
+```js title="astro.config.mjs" del={4}
+import { defineConfig } from "astro/config";
+
+export default defineConfig({
+	compressHTML: true
+})
+```
+
+You must now set `compressHTML: false` to opt out of HTML compression.
+
+### Changed default: `scopedStyleStrategy`
+
+In Astro v2.x, the default value of [`scopedStyleStrategy`](/en/reference/configuration-reference/#scopedstylestrategy) was `"where"`.
+
+Astro v3.0 introduces a new, default value: `"attribute"`. By default, styles are now applied using `data-*` attributes.
+
+#### What should I do?
+
+To retain your project's current [style scoping](/en/guides/styling/#scoped-styles), update the configuration file to the previous default value: 
+
+
+```js title="astro.config.mjs" ins={4}
+import { defineConfig } from "astro/config";
+
+export default defineConfig({
+	scopedStyleStrategy: "where"
+})
+```
+
+### Changed default: `inlineStyleSheets`
+
+In Astro v2.x, all project stylesheets were sent as link tags by default. You could opt in to `"always"` inlining them into `<style>` tags, or to `"auto"`matically inlining only stylesheets below a certain size by setting the [`build.inlineStylesheets`](/en/reference/configuration-reference/#buildinlinestylesheets) configuration. The default setting was `"never"`.
+
+Astro v3.0 changes the default value of `inlineStylesheets` to `"auto"`. Stylesheets smaller than `ViteConfig.build.assetsInlineLimit` (default: 4kb) are inlined by default. Otherwise, project styles are sent in external stylesheets.
+
+#### What should I do?
+If you want to keep your project's current behavior, set `build.inlineStylesheets` to the previous default, `"never"`:
+
+
+```js title="astro.config.mjs" ins={4-6}
+import { defineConfig } from "astro/config";
+
+export default defineConfig({
+	 build: {
+    inlineStylesheets: "never"
+  }
+})
+```
+
+### Changed default: image service
+
+In Astro v2.x, Squoosh was the [default image processing service](/en/guides/images/#default-image-service).
+
+Astro v3.0 now includes Sharp as the default image processing service, and instead provides a configuration option to use Squoosh.
+
+#### What should I do?
+
+If you had previously installed Sharp, uninstall the dependency using your package manager, or by removing manually from your `package.json`. It is now included in Astro.
```

Comment:
I don't believe removal of `Sharp` should be recommended. See https://github.com/withastro/astro/issues/7966 and related issue.

Reply:
Thanks for pointing this out, @fflaten ! Would you like to suggest a change in wording here? Do you think it's safe to say "you MAY be able to remove Sharp", or would you instead suggest NO mention at all?

Also asking @Princesseuh for her advice here. Is there any harm in having this as a separate, manually installed dependency if Astro DOES successfully do it for you?

Happy to let you two figure out the best advice here, then I'll just commit it!

Reply:
I'm very new to web dev, but AFAIK there's no clear downside with keeping it as a direct dependency. If true, then I'd leave it out.

Similarly I'd keep it out of install docs and let the error message help any affected users.

Reply:
No downside keeping it, in fact `pnpm` users even need it directly installed in their projects, so I'd say might as well.

The only potential downside is their sharp install being duped, but we try to maintain a good enough semver range that this shouldn't happen.